### PR TITLE
CORDA-2893 Upgrade Corda to use Gradle 5.x

### PR DESCRIPTION
**Pull request by josecoll**
**NOTE: dependent on publication of Corda Gradle Plugins 5.0.0 in https://github.com/corda/corda-gradle-plugins/pull/209**

https://r3-cev.atlassian.net/browse/CORDA-2893 

Dependent on following changes:

In Corda Gradle plugins V5.0.0:
- [Fix API Scanner Gradle plugin to work in Gradle 5.x](https://r3-cev.atlassian.net/browse/CORDA-2889). 
- [Fix CorDapp plugin also to work with Gradle 5.2 and above.](https://r3-cev.atlassian.net/browse/CORDA-2896)

Other:
- Upgrade Capsule Gradle plugin to work in Gradle 5.x.
   This required forking the original [danthegoodman repository](https://github.com/danthegoodman/gradle-capsule-plugin) into a [Corda repository](https://github.com/corda/gradle-capsule-plugin) and applying a local upgrade.

### CODE REVIEW

File: common/logging/build.gradle [hunk 1]
```diff
@@ -8,12 +8,12 @@ dependencies {
     compile group: "org.jetbrains.kotlin", name: "kotlin-reflect", version: kotlin_version
 
     compile group: "com.typesafe", name: "config", version: typesafe_config_version
-    compile "org.apache.logging.log4j:log4j-slf4j-impl:$log4j_version"
 
+    // Log4J: logging framework
+    compile "org.apache.logging.log4j:log4j-core:$log4j_version"
```
Comment (chrisr3) [inline common/logging/build.gradle]: Is this correct? Officially, our logging API is SLF4J. Log4J2 is just our _particular_ SLF4J implementation.

File: settings.gradle [hunk 2]
```diff
@@ -1,3 +1,13 @@
+pluginManagement {
+    ext.artifactory_contextUrl = 'https://ci-artifactory.corda.r3cev.com/artifactory'
+
+    repositories {
+        mavenLocal()
+        maven { url "$artifactory_contextUrl/corda-dev" }
+        maven { url "$artifactory_contextUrl/corda-dependencies" }
+        gradlePluginPortal()
```
Comment (chrisr3) [inline settings.gradle]: I think the `gradlePluginPortal()` should be firs or second,t because that's still where we expect to find the vast marjority of our Gradle plugins. And we'd only expect to find local `-SNAPSHOT` builds of our own plugins in `mavenLocal()`.

File: tools/demobench/build.gradle [hunk 3]
```diff
@@ -66,6 +66,7 @@ dependencies {
 
     // FontAwesomeFX: icons in the form of a font.
     compile "de.jensd:fontawesomefx-fontawesome:4.7.0-5"
+    compile 'de.jensd:fontawesomefx-commons:8.15'
```
Comment (chrisr3) [inline tools/demobench/build.gradle]: Why is this needed?

File: tools/explorer/build.gradle [hunk 4]
```diff
@@ -30,6 +30,7 @@ dependencies {
 
     // FontAwesomeFX: The "FontAwesome" icon library.
     compile 'de.jensd:fontawesomefx-fontawesome:4.7.0'
+    compile 'de.jensd:fontawesomefx-commons:8.13'
```
Comment (chrisr3) [inline tools/explorer/build.gradle]: Similarly - why is this needed?

File: settings.gradle (hunk 2 above)
Comment (josecoll) [reply settings.gradle]: I recommend a follow-up PR be raised to update all references from 
https://ci-artifactory.corda.r3cev.com
to
https://software.r3.com

File: tools/demobench/build.gradle (hunk 3 above)
Comment (josecoll) [reply tools/demobench/build.gradle]: code will not compile without

File: tools/explorer/build.gradle (hunk 4 above)
Comment (josecoll) [reply tools/explorer/build.gradle]: ditto

File: settings.gradle (hunk 2 above)
Comment (josecoll) [reply settings.gradle]: I will adjust the plugins declaration as per your suggestion.

Comment (chrisr3) [reply settings.gradle]: Why is `mavenLocal()` declared last here?

Comment (josecoll) [reply settings.gradle]: Damn! stupid me. Fixed.
